fractal_server/app/routes/api/v1/project.py

```diff
@@ -430,6 +431,37 @@ async def apply_workflow(
         ),
         **apply_workflow.dict(),
     )
+
+    # Rate Limiting:
+    # raise `429 TOO MANY REQUESTS` if this endpoint has been called with the
+    # same database keys (Project, Workflow and Datasets) during the last
+    # `settings.FRACTAL_API_SUBMIT_RATE_LIMIT` seconds.
+    stm = (
+        select(ApplyWorkflow)
+        .where(ApplyWorkflow.project_id == project_id)
+        .where(ApplyWorkflow.workflow_id == workflow_id)
+        .where(ApplyWorkflow.input_dataset_id == input_dataset_id)
+        .where(ApplyWorkflow.output_dataset_id == output_dataset_id)
+        .order_by(ApplyWorkflow.start_timestamp.desc())
+        .limit(1)
```

Comment:
Should we also include some of the `**apply_workflow.dict()` attributes in this check?